llvm/lib/MC/DXContainerRootSignature.cpp

```diff
@@ -7,17 +7,84 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/MC/DXContainerRootSignature.h"
+#include "llvm/ADT/SmallString.h"
 #include "llvm/Support/EndianStream.h"
 
 using namespace llvm;
 using namespace llvm::mcdxbc;
 
+static uint32_t writePlaceholder(raw_svector_ostream &Stream) {
+  const uint32_t DummyValue = std::numeric_limits<uint32_t>::max();
+  uint32_t Offset = Stream.tell();
+  support::endian::write(Stream, DummyValue, llvm::endianness::little);
+  return Offset;
+}
+
+static void rewriteOffsetToCurrentByte(raw_svector_ostream &Stream,
+                                       uint32_t Offset) {
+  uint32_t Value =
+      support::endian::byte_swap<uint32_t, llvm::endianness::little>(
+          Stream.tell());
+  Stream.pwrite(reinterpret_cast<const char *>(&Value), sizeof(Value), Offset);
+}
+
+size_t RootSignatureDesc::getSize() const {
+  size_t Size = sizeof(dxbc::RootSignatureHeader) +
+                Parameters.size() * sizeof(dxbc::RootParameterHeader);
+
+  for (const auto &P : Parameters) {
```

Comment:
Best to only use auto when the type is somewhere in the expression (or it's something obvious but long like an iterator)
```suggestion
  for (const mcdxbc::RootParameter &P : Parameters) {
```